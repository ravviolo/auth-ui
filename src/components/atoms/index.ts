export { Button } from './Button/Button';
export { Checkbox } from './Checkbox/Checkbox';
export { Divider } from './Divider/Divider';
export { Heading } from './Heading/Heading';
export { IconButton } from './IconButton/IconButton';
export { InputField } from './InputField/InputField';
export { InputLabel } from './InputLabel/InputLabel';
export { TextButton } from './TextButton/TextButton';
