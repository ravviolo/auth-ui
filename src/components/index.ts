export * from './templates/LoginTemplate/LoginTemplate';
export * from './templates/SignUpTemplate/SignUpTemplate';
