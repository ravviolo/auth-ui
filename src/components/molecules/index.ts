export { AuthFooter } from './AuthFooter/AuthFooter';
export { AuthHeader } from './AuthHeader/AuthHeader';
export { AuthSocials } from './AuthSocials/AuthSocials';
export { EmailInput } from './EmailInput/EmailInput';
export { PasswordInput } from './PasswordInput/PasswordInput';
