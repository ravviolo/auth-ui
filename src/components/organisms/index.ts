export * from './LoginForm/LoginForm';
export * from './SignUpForm/SignUpForm';
