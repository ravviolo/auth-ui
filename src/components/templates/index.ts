export * from './LoginTemplate/LoginTemplate';
export * from './SignUpTemplate/SignUpTemplate';
