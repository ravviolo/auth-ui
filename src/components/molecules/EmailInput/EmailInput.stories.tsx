import { ComponentMeta, ComponentStory } from '@storybook/react';

import { EmailInput } from 'components/molecules';

export default {
  title: 'Molecules/EmailInput',
  component: EmailInput,
  args: {
    id: 'email',
    name: 'email',
  },
} as ComponentMeta<typeof EmailInput>;

export const Default: ComponentStory<typeof EmailInput> = (args) => (
  <EmailInput {...args} />
);
Default.storyName = 'EmailInput';
