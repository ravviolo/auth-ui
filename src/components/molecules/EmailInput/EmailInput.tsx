import React from 'react';

import { InputField, InputLabel } from 'components/atoms';

import * as S from './EmailInput.styles';

export interface Props {
  id: string;
  name: string;
  testId: string;
}

export const EmailInput = React.forwardRef<HTMLInputElement, Props>(
  ({ id, name, testId }, ref) => {
    return (
      <S.InputContainer direction="column" testId={testId}>
        <InputLabel
          htmlFor={id}
          label="Email"
          testId="input-label-email-test-id"
        />
        <InputField
          ref={ref}
          id={id}
          name={name}
          testId="input-field-email-test-id"
          type="email"
        />
      </S.InputContainer>
    );
  }
);

EmailInput.displayName = 'EmailInput';
