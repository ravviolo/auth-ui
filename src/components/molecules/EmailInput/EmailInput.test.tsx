import { render, screen, within } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { create } from 'react-test-renderer';

import { EmailInput } from 'components/molecules';

import { Props as EmailInputProps } from './EmailInput';

describe('EmailInput', () => {
  const props: EmailInputProps = {
    id: 'test-email',
    name: 'test-email',
    testId: 'email-input-test-id',
  };

  it('should render input of type email', () => {
    render(<EmailInput {...props} />);

    const emailInput = screen.getByTestId('email-input-test-id');
    const emailInputField = within(emailInput).getByTestId(
      'input-field-email-test-id'
    );

    expect(emailInputField).toHaveAttribute('type', 'email');
  });

  it('should correctly assign id, name attributes to input field', () => {
    render(<EmailInput {...props} />);

    const emailInput = screen.getByTestId('email-input-test-id');
    const emailInputField = within(emailInput).getByTestId(
      'input-field-email-test-id'
    );

    expect(emailInputField).toHaveAttribute('id', 'test-email');
    expect(emailInputField).toHaveAttribute('name', 'test-email');
  });

  it('should semantically connect input label with input field', () => {
    render(<EmailInput {...props} />);

    const emailInput = screen.getByTestId('email-input-test-id');
    const emailInputLabel = within(emailInput).getByTestId(
      'input-label-email-test-id'
    );
    const emailInputField = within(emailInput).getByTestId(
      'input-field-email-test-id'
    );

    expect(emailInputLabel).toHaveAttribute(
      'for',
      emailInputField.getAttribute('id')
    );
  });

  it('should render initially empty input', () => {
    render(<EmailInput {...props} />);

    const emailInput = screen.getByTestId('email-input-test-id');
    const emailInputField = within(emailInput).getByTestId(
      'input-field-email-test-id'
    );

    expect(emailInputField).toHaveValue('');
  });

  it("should display 'Email' as input label", () => {
    render(<EmailInput {...props} />);

    const emailInput = screen.getByTestId('email-input-test-id');
    const emailInputLabel = within(emailInput).getByTestId(
      'input-label-email-test-id'
    );

    expect(emailInputLabel).toHaveTextContent('Email');
  });

  it('should display entered text in input field', async () => {
    render(<EmailInput {...props} />);

    const emailInput = screen.getByTestId('email-input-test-id');

    const emailInputField = within(emailInput).getByTestId(
      'input-field-email-test-id'
    );

    await userEvent.type(emailInputField, 'Testing input');

    expect(emailInputField).toHaveValue('Testing input');
  });

  it('should match snapshot', () => {
    const tree = create(<EmailInput {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
