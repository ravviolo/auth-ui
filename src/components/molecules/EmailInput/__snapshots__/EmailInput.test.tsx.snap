// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailInput should match snapshot 1`] = `
<div
  className="sc-jSMfEi bQrDqy sc-breuTD jjaaRC"
  data-testid="email-input-test-id"
  direction="column"
>
  <label
    className="sc-papXJ daxRFt"
    data-testid="input-label-email-test-id"
    htmlFor="test-email"
  >
    Email
  </label>
  <input
    className="sc-ftvSup dLJLzu"
    data-testid="input-field-email-test-id"
    id="test-email"
    name="test-email"
    type="email"
  />
</div>
`;
